jum.py

```diff
@@ -35,48 +35,26 @@ def __init__(self):
         self.smiley = utils.load_image('smiley.png', True)
 
     def setup(self):
-        self.obj_grid = []
-        for i in range(10000):
-            r = random.randint(1, self.n)
-            if r not in self.obj_grid:
-                self.obj_grid.append(r)
-            if len(self.obj_grid) == self.total:
-                break
-        self.obj_bgd = []
-        for i in range(1, self.n + 1):
-            if i not in self.obj_grid:
-                self.obj_bgd.append(i)
+        self.obj_grid = list(range(1, self.n + 1))

```

Comment:
These statements makes a list of self.n items, then shuffles them, then truncates the list to self.total.  It is very different.  How do you know it is more efficient?

Reply:
Thanks for the explanation, but please add it to the commit message.  We do not keep pull request comments in git.

Even so, n is 83.  Why code for n of 83000 when n is 83?  The return on invested time seems poor, especially when balanced with risk of new bugs.

Reply:
That loop is actually called for 2 seconds on every shuffle, and FPS=40. So it runs 80 times every shuffle... The code snippet above tested if it ran 100 times. So it is still pretty significant.

Reply:
Is a delay now needed to ensure an animation works as before?

Reply:
Um, `setup` function is called 80 times to place images randomly which gives that shuffling animation effect. So we do need to call it multiple times.

Reply:
What is the change to the animation effect?  How do we maintain visualisation?

Reply:
I didn't change any animations and game logic is still the same. Though I plan to improve on animations in future